bigtable/google/cloud/bigtable/table.py

```diff
@@ -363,6 +375,74 @@ def sample_row_keys(self):
         return response_iterator
 
 
+class _MutateRowsRetryableError(Exception):
+    """A retryable error in Mutate Rows response."""
+    pass
+
+
+class _RetryableMutateRowsWorker(object):
+    # pylint: disable=unsubscriptable-object
+    RETRY_CODES = (
+        StatusCode.DEADLINE_EXCEEDED.value[0],
+        StatusCode.ABORTED.value[0],
+        StatusCode.UNAVAILABLE.value[0],
+    )
+
+    def __init__(self, client, table_name, rows):
+        self.client = client
+        self.table_name = table_name
+        self.rows = rows
+        self.responses_statuses = [
+            None for _ in six.moves.xrange(len(self.rows))]
+
+    def __call__(self):
+        return self._do_mutate_retryable_rows()
+
+    def _is_retryable(self, status):  # pylint: disable=no-self-use
+        return (status is None or
+                status.code in _RetryableMutateRowsWorker.RETRY_CODES)
+
+    def _next_retryable_row_index(self, begin_index):
+        i = begin_index
+        while i < len(self.responses_statuses):
+            status = self.responses_statuses[i]
+            if self._is_retryable(status):
+                return i
+            i += 1
+        return -1
+
+    def _do_mutate_retryable_rows(self):
+        retryable_rows = []
+        for i, status in enumerate(self.responses_statuses):
+            if self._is_retryable(status):
+                retryable_rows.append(self.rows[i])
+
+        if not retryable_rows:
+            # All mutations are either successful or non-retryable now.
+            return self.responses_statuses
+
+        mutate_rows_request = _mutate_rows_request(
+            self.table_name, retryable_rows)
+        responses = self.client._data_stub.MutateRows(
+            mutate_rows_request)
+
+        num_retryable_responses = 0
```

Comment:
Would be good to do a sanity check that # rows == # responses

Reply:
Do you mean `assert`? 
I thought of a `raise` as well but that means we have to update the `table.mutate_rows` API doc.

Reply:
I think assert would do.

Reply:
Just saw your reply. Done in https://github.com/GoogleCloudPlatform/google-cloud-python/pull/4116/commits/184efbeebd1acd1ae57eac7ecfd372235fac3b32